pkg/services/ngalert/store/alert_rule.go

```diff
@@ -434,58 +433,34 @@ func (st DBstore) GetRuleGroupInterval(ctx context.Context, orgID int64, namespa
 	})
 }
 
-// GetUserVisibleNamespaces returns the folders that are visible to the user and have at least one alert in it
+// GetUserVisibleNamespaces returns the folders that are visible to the user
 func (st DBstore) GetUserVisibleNamespaces(ctx context.Context, orgID int64, user identity.Requester) (map[string]*folder.Folder, error) {
-	namespaceMap := make(map[string]*folder.Folder)
-
-	searchQuery := dashboards.FindPersistedDashboardsQuery{
-		OrgId:        orgID,
+	folders, err := st.FolderService.GetFolders(ctx, folder.GetFoldersQuery{
+		OrgID:        orgID,
+		WithFullpath: true,
 		SignedInUser: user,
-		Type:         searchstore.TypeAlertFolder,
-		Limit:        -1,
-		Permission:   dashboardaccess.PERMISSION_VIEW,
-		Sort:         model.SortOption{},
-		Filters: []any{
-			searchstore.FolderWithAlertsFilter{},
-		},
+	})
+	if err != nil {
+		return nil, err
 	}
 
-	var page int64 = 1
-	for {
-		query := searchQuery
-		query.Page = page
-		proj, err := st.DashboardService.FindDashboards(ctx, &query)
-		if err != nil {
-			return nil, err
-		}
-
-		if len(proj) == 0 {
-			break
-		}
-
-		for _, hit := range proj {
-			if !hit.IsFolder {
-				continue
-			}
-			namespaceMap[hit.UID] = &folder.Folder{
-				UID:       hit.UID,
-				Title:     hit.Title,
-				ParentUID: hit.FolderUID,
-			}
-		}
-		page += 1
+	namespaceMap := make(map[string]*folder.Folder)
+	for _, f := range folders {
+		namespaceMap[f.UID] = f
 	}
 	return namespaceMap, nil
 }
 
 // GetNamespaceByUID is a handler for retrieving a namespace by its UID. Alerting rules follow a Grafana folder-like structure which we call namespaces.
 func (st DBstore) GetNamespaceByUID(ctx context.Context, uid string, orgID int64, user identity.Requester) (*folder.Folder, error) {
-	folder, err := st.FolderService.Get(ctx, &folder.GetFolderQuery{OrgID: orgID, UID: &uid, SignedInUser: user})
+	f, err := st.FolderService.GetFolders(ctx, folder.GetFoldersQuery{OrgID: orgID, UIDs: []string{uid}, WithFullpath: true, SignedInUser: user})
 	if err != nil {
 		return nil, err
 	}
-
-	return folder, nil
+	if len(f) == 0 {
+		return nil, dashboards.ErrFolderAccessDenied
+	}
```

Comment:
This is to replicate `FolderService.Get` behavior.

Reply:
sure, I'll do but I would suggest not to block the current work (merge this one and address it in a follow up)

Reply:
@yuri-tceretian the `FolderService.Get()` change: #81972